File: Cola_Lista/ColaLista.java
package estructura_de_datos.Cola_Lista;

import java.util.Scanner;

class Nodo {
    int dato;
    Nodo sig;
    Nodo(int d) { dato = d; sig = null; }
}

public class ColaLista {
    static Nodo frente = null, fin = null;
    static Scanner sc = new Scanner(System.in);

    static void insertar() {
        System.out.print("\nIngrese el elemento: ");
        int elem = sc.nextInt();
        Nodo nuevo = new Nodo(elem);

        if (frente == null) frente = fin = nuevo;
        else {
            fin.sig = nuevo;
            fin = nuevo;
        }
        System.out.println("Elemento insertado correctamente.");
    }

    static void eliminar() {
        if (frente == null) {
            System.out.println("\nSUBDESBORDAMIENTO (UNDERFLOW)");
            return;
        }
        int e = frente.dato;
        frente = frente.sig;
        if (frente == null) fin = null;
        System.out.println("\nElemento eliminado: " + e);
    }

    static void mostrar() {
        if (frente == null) {
            System.out.println("\nLa cola está vacía.");
            return;
        }
        System.out.println("\nElementos en la cola:");
        for (Nodo i = frente; i != null; i = i.sig)
            System.out.println(i.dato);
    }

    public static void main(String[] args) {
        int op;
        do {
            System.out.println("\n*************** MENÚ PRINCIPAL ***************");
            System.out.println("1. Insertar un elemento");
            System.out.println("2. Eliminar un elemento");
            System.out.println("3. Mostrar la cola");
            System.out.println("4. Salir");
            System.out.print("Ingrese su opción: ");
            op = sc.nextInt();

            switch (op) {
                case 1: insertar(); break;
                case 2: eliminar(); break;
                case 3: mostrar(); break;
                case 4: System.out.println("\nSaliendo..."); break;
                default: System.out.println("\nOpción inválida."); break;
            }
        } while (op != 4);
    }
}

File: Cola_Lista/ColaLista.cs
using System;

class Nodo {
    public int dato;
    public Nodo sig;
    public Nodo(int d) { dato = d; sig = null; }
}

class ColaLista {
    static Nodo frente = null, fin = null;

    static void Insertar() {
        Console.Write("\nIngrese el elemento: ");
        int elem = int.Parse(Console.ReadLine());
        Nodo nuevo = new Nodo(elem);

        if (frente == null) frente = fin = nuevo;
        else {
            fin.sig = nuevo;
            fin = nuevo;
        }
        Console.WriteLine("Elemento insertado correctamente.");
    }

    static void Eliminar() {
        if (frente == null) {
            Console.WriteLine("\nSUBDESBORDAMIENTO (UNDERFLOW)");
            return;
        }
        int e = frente.dato;
        frente = frente.sig;
        if (frente == null) fin = null;
        Console.WriteLine("\nElemento eliminado: " + e);
    }

    static void Mostrar() {
        if (frente == null) {
            Console.WriteLine("\nLa cola está vacía.");
            return;
        }
        Console.WriteLine("\nElementos en la cola:");
        for (Nodo i = frente; i != null; i = i.sig)
            Console.WriteLine(i.dato);
    }

    static void Main() {
        int op;
        do {
            Console.WriteLine("\n*************** MENÚ PRINCIPAL ***************");
            Console.WriteLine("1. Insertar un elemento");
            Console.WriteLine("2. Eliminar un elemento");
            Console.WriteLine("3. Mostrar la cola");
            Console.WriteLine("4. Salir");
            Console.Write("Ingrese su opción: ");
            op = int.Parse(Console.ReadLine());

            switch(op) {
                case 1: Insertar(); break;
                case 2: Eliminar(); break;
                case 3: Mostrar(); break;
                case 4: Console.WriteLine("\nSaliendo..."); break;
                default: Console.WriteLine("\nOpción inválida."); break;
            }
        } while (op != 4);
    }
}
